deploy: remove external-attacher sidecar from cephfs deployment

humblec opened the pull request:
CephFS CSI driver dont have/advertise controller publish/unpublish capabilities, thus dont need attacher sidecar for its operations.
Also this removal take care of the burden on unncessary operation (round-trip) in the preparation of a volume for a container, and requires CSI Drivers to deploy an unnecessary sidecar container (external-attacher).

Things covered in this PR 
- [x]  Removed attacher sidecar from deploy yamls and also from Helm
- [x] Removed/adjusted RBAC supporting this removal
- [x] CephFS CSI driver object have been changed to `attachRequired:false` param as attachment is no longer required.  

Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

Madhu-1 commented on deploy/cephfs/kubernetes/csi-cephfsplugin-provisioner.yaml:
```diff
@@ -90,60 +90,6 @@ spec:
-          volumeMounts:
-            - name: socket-dir
-              mountPath: /csi
-        - name: csi-cephfsplugin
```
Even if its a WIP why cephfsplugin is removed?

humblec committed:
deploy: remove external-attacher sidecar from cephfs deployment

CephFS CSI driver dont need attacher sidecar for its operations.
This commit remove the same. The RBAC has also got adjusted.

Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

nixpanic commented:
> CephFS CSI driver dont need attacher sidecar for its operations.

It helps if you explain the **why**.

humblec commented:
> > CephFS CSI driver dont need attacher sidecar for its operations.
> 
> It helps if you explain the **why**.

Sure . Updated the details in the PR description.

humblec commented:
@Madhu-1 @nixpanic comments are addressed.. ptal.. thanks!

Madhu-1 commented:
As mentioned here https://github.com/ceph/ceph-csi/issues/1106#issuecomment-634424544 This was already attempted a couple of times https://github.com/ceph/ceph-csi/issues/1106#issuecomment-634659533, Are we sure removing is not having any problem. Have we don't all required testing to make sure no problem occurs if we remove it?

humblec commented:
> As mentioned here [#1106 (comment)](https://github.com/ceph/ceph-csi/issues/1106#issuecomment-634424544) This was already attempted a couple of times [#1106 (comment)](https://github.com/ceph/ceph-csi/issues/1106#issuecomment-634659533), Are we sure removing is not having any problem. Have we don't all required testing to make sure no problem occurs if we remove it?

@Madhu-1 As mentioned in the reply thread https://github.com/ceph/ceph-csi/issues/1106#issuecomment-875737624, it should be fine to remove .. I dont expect any issues due to the same.  More or less, many CSI drivers dont use this sidecar at all as it is unwanted. so we are good :+1:

humblec committed:
deploy: remove `attachrequired` param from csidriver object

As the attacher is no longer required we have to mention the same
for csidriver object parameter.

Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

humblec committed:
doc: remove attacher sidecar mentions from chart README

This commit remove the references to the attacher sidecar
attributes.

Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

humblec commented:
/retest ci/centos/mini-e2e/k8s-1.23

humblec commented:
/retest ci/centos/mini-e2e/k8s-1.23

Madhu-1 reviewed (commented):
@humblec please squash the commits into one or 2 IMO we don't need to have many commits to for this one. and also retest can be done once PR gets approval.

humblec commented:
> @humblec please squash the commits into one or 2 IMO we don't need to have many commits to for this one. and also retest can be done once PR gets approval.

@Madhu-1 Done.. ptal.. thanks!

humblec commented:
/retest ci/centos/mini-e2e-helm/k8s-1.21

humblec commented:
/retest ci/centos/mini-e2e-helm/k8s-1.22

humblec commented:
@Madhu-1 comments are addresssed. ptal. thanks..

Madhu-1 reviewed (approved):
There was no need to have a different commit for the Doc. ideally all changes should go into 1 or 2 commits.

Considering all the tests done as per https://github.com/ceph/ceph-csi/pull/3149#issuecomment-1143559182 to make sure we don't need an attacher anymore approving the PR.

Madhu-1 commented on deploy/cephfs/kubernetes/csi-cephfsplugin-provisioner.yaml:
```diff
@@ -90,20 +90,6 @@ spec:
           volumeMounts:
             - name: socket-dir
               mountPath: /csi
-        - name: csi-cephfsplugin-attacher
```
please remove the RBAC for this one in the deploy/cephfs folder.

humblec replied:
@Madhu-1 done. thanks.. ptal.

Madhu-1 reviewed (changes requested):
looks like RBAC for kubernetes templates is not removed yet.

humblec commented:
/retest ci/centos/mini-e2e-helm/k8s-1.21

humblec commented:
@Mergifyio refresh

thrashwerk commented:
I'm running a k8s 1.24 cluster and I've got a few static CephFS filesystems for which I have created static PVs and PVCs manually and use the PVCs to mount CephFS to pods.

After updating to 3.7.0 new pods just get stuck with an event `Unable to attach or mount volumes: unmounted volumes=[ceph-pvc-storage1], unattached volumes=[kube-api-access-h2rfp ceph-pvc-storage1]: timed out waiting for the condition`.

I don't get how things are supposed to work now. Why doesn't the CephFS PV get attached?

Madhu-1 commented:
@thrashwerk can you please open a new issue?
